### DISCUSSION TITLE
CrUX Top 1M

### DISCUSSION THREAD

**Discussion by romain-fontugne** [InternetHealthReport/internet-yellow-pages#153] [New dataset] [answered]
# Description

Google provides a list of top websites visited by Chrome user:  [Chrome UX Report](https://developer.chrome.com/docs/crux). One advantage of CrUX compared to the Tranco top1M is that they also provide ranking per countries.

Data is updated monthly.

# Modeling

No new nodes are required.
They technically provide a web-origin (e.g. https://www.google.com) but we can simply extract the hostname part (e.g. www.google.com). 

For the country rankings:
```cypher
(:HostName)-[:RANK {rank: 1, country: 'JP', protocol: 'https'}]->(:Ranking {name: 'CrUX top 1M (JP)'})-[:COUNTRY]->(:Country)
```

For the global ranking:
```cypher
(:HostName)-[:RANK {rank: 1, country: 'JP', protocol: 'https'}]->(:Ranking {name: 'CrUX top 1M'})
```
# Comments
The data is available on github:
- https://github.com/zakird/crux-top-lists
- https://github.com/InternetHealthReport/crux-top-lists-country

Next steps are too ensure that we get IP addresses for all these domains (maybe via OpenINTEL). Also discussed here: https://github.com/InternetHealthReport/iyp-notebooks/issues/1

**Comment by m-appel** [comment, marked as answer]
Implemented in #160.
